TakingITGlobal/commit2act
# [TIG-695] Progress provider for steps

## Ticket

* [TIG-695](https://app.clickup.com/t/9009201449/TIG-695)

## Description

This PR primarily adds the `useSignupStepProgress` hook which pulls the `step` and `progress` values from `useSignupPageRoute` (if available) and returns those along with formatted label strings. Since the hook returns both the `step` and `progress` numbers, we're removing the `step` prop from `SignupContainer` and grabbing that data directly from the hook within that component. This also removes the use of `useSignupPageRoute` and manually passing the `step` within each of the signup page components.

The approach used here prevents any context store and provider wrapper from being used across pages - leaving fewer opportunities for complexities and issues. Either the `progress` number or `progressLabel` string can be used with CSS to style the progress bar component.

~~**Note:** This branch can not be merged until [TIG-725](https://github.com/TakingITGlobal/commit2act/pull/107) is reviewed, merged into `feature/sign-up-flow`, and rebased on this branch.~~ Done!

## Testing

* Pull down and install the branch
* Run `npm run test` and confirm all tests pass
* Run `npm run storybook` and confirm all stories under Pages > Sign Up render with the progress percentage
* Run `npm run start` and visit /account/create once the app loads in browser
* Confirm that each page in the flow shows the accurate progress percentage above the heading

## Screenshots

**Percentage visible for the 4th step's Storybook entry:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/ea360e32-553d-47f9-8c90-0419fde299af)

**Last step with 100% progress shown:**

![image](https://github.com/TakingITGlobal/commit2act/assets/1672105/99485fd6-a76d-4989-a373-ee1c12c0fcba)
